lib/src/rules/directives_ordering.dart

```diff
@@ -199,6 +247,31 @@ class _Visitor extends SimpleAstVisitor {
         .forEach(reportDirective);
   }
 
+  void _checkDirectiveSectionOrderedAlphabetically(
+      Set<AstNode> lintedNodes, CompilationUnit node) {
+    final importDirectives = _getImportDirectives(node);
+    final exportDirectives = _getExportDirectives(node);
+
+    final dartImports = importDirectives.where(_isDartImport);
+    final relativeImports = importDirectives.where(_isRelativeImport);
+
+    _checkSectionInOrder(lintedNodes, dartImports);
+    _checkSectionInOrder(lintedNodes, relativeImports);
+    _checkSectionInOrder(lintedNodes, exportDirectives);
+
+    if (project != null) {
```

Comment:
Can the project actually ever be null? (I honestly don't know, so this isn't a trick question.) If it can, should we check the order of all `package:` imports, ignoring the distinction between "own" and "third party", or would that sometimes lead to cases where we report lints in code that is actually correctly ordered?